File: Implementierung/FrontendAPI/keys.py
import os
from pathlib import Path
from Implementierung.FrontendAPI import utilities

user_status_name: str = "userStatus"
user_privilege_name: str = "userPrivilege"
password_name: str = "password"
repeat_password_name: str = "repeatPassword"
user_name = "userName"

template_name: str = "templateName"
workflow_instance_name: str = "workflowInstanceName"
version_number_name: str = "versionNumber"
version_note_name: str = "versionNote"
config_file_name: str = "configFileName"
versions_name: str = "versions"
key_value_pairs_name: str = "keyValuePairs"

workflow_instance_names: str = "workflowInstanceNames"
config_file_names: str = "configFileNames"
template_names: str = "templateNames"

dag_picture_name: str = "dagPicture"
dag_definition_name: str = "dagDefinitionFile"

status_code_name: str = "statusCode"


server_name: str = "serverName"
selected_for_execution_name: str = "selectedForExecution"
server_resources_name: str = "serverResources"
server_address_name: str = "serverAddress"
server_status_name: str = "serverStatus"
container_limit_name: str = "containerLimit"


temp_in_name: str = "temp_in"
temp_out_name: str = "temp_out"

files_key: str = "file[]"
file_key: str = "file"

config_save_path: str = "config"
dag_save_path: str = "dag_file"

underscore: str = "_"

temp_in_path: str = os.path.join(utilities.parent_path, temp_in_name)

File: matflow/frontend/src/Controler/Keys.ts
export const keys = {
    userStatusName: "userStatus",
    userPrivilegeName: "userPrivilege",
    passwordName: "password",
    repeatPasswordName: "repeatPassword",
    userName: "userName",
    allUsers: "users",

    templateName: "templateName",
    workflowInstanceName: "workflowInstanceName",
    versionNumberName: "versionNumber",
    versionNoteName: "versionNote",
    configFileName: "configFileName",
    versionsName: "versions",
    frontendVersionsChanges: "parameterChanges",
    keyValuePairsName: "keyValuePairs",

    workflowInstanceNames: "workflowInstanceNames",
    configFileNames: "configFileNames",
    templateNames: "templateNames",
    configFiles: "configFiles",

    namesAndConfigs: "workflowInstanceConfigNames",

    dagPictureName: "dagPicture",
    dagDefinitionName: "dagDefinitionFile",

    statusCodeName: "statusCode",


    serverName: "serverName",
    selectedForExecutionName: "selectedForExecution",
    serverResourcesName: "serverResources",
    serverAddressName: "serverAddress",
    serverStatusName: "serverStatus",
    containerLimitName: "containerLimit",


    tempInName: "temp_in",
    tempOutName: "temp_out",

    filesKey: "file[]",
    fileKey: "file",

    configSavePath: "config",
    dagSavePath: "dag_file",

    underscore: "_",

    oldKey: "oldKey",
    newKey: "newKey",
    oldValue: "oldValue",
    newValue: "newValue",

    // api calls names
    getServerDetails: "/get_server_details",
    setServerDetails: "/set_server_details",
    getAllUsersAndDetails: "/get_all_users_and_details",
    setUserDetails: "/set_user_details",
    deleteUser: "/delete_user",
    getWfInstanceVersions: "/get_wf_instance_versions",
    replaceWfInstanceVersion: "/replace_wf_instance_active_version",
    createVersionOfWfInstance: "/create_version_of_wf_instance",
    getConfigFromWfInstance: "/get_config_from_wf_instance",
    createWfInstance: "/create_workflow_instance",
    getAllWfInstancesNamesAndConfigFileNames: "/get_all_wf_instances_names_and_config_file_names",
    registerUser: "/register_user",
    createTemplate: "/create_template",
    getAllTemplateNames: "/get_all_template_names",
    getTemplate: "/get_template",
    getGraphForTemporaryTemplate: "/get_graph_for_temporary_template",

} as const;